packages/fxa-auth-server/lib/routes/password.js

```diff
@@ -757,13 +758,68 @@ module.exports = function (
             });
           })
           .then(() =>
-            request.emitMetricsEvent('password.forgot.verify_code.completed')
+            request.emitMetricsEvent('password.forgot.verify_code.completed'),
           )
           .then(() => ({
             accountResetToken: accountResetToken.data,
           }));
       },
     },
+    {
+      method: 'POST',
+      path: '/password/create',
+      options: {
+        ...PASSWORD_DOCS.PASSWORD_CREATE_POST,
+        auth: {
+          strategy: 'sessionToken',
+        },
+        response: {
+          schema: isA
+            .object({
+              authPW: isA.string(),
+            }),
+        },
+      },
+      handler: async function(request) {
+        log.begin('Password.create', request);
+        const sessionToken = request.auth.credentials;
+        const { uid } = sessionToken;
+
+        const { authPW } = request.payload;
+        
+        const account = await db.account(uid);
+        // We don't allow users that have a password set already to create a new password
+        // because this process would destroy their original encryption keys and might
+        // leave the account in an invalid state.
+        if (account.verifierSetAt > 0) {
+          throw error.cannotCreatePassword();
+        }
+        
+        // Users that have enabled 2FA must be in a 2FA verified session to create a password.
```

Comment:
While this is impossible to do in the current UX, it is possible if using our api directly.

Reply:
Yeah agreed we should have this guard, good call 👍 Also thanks for the comments.